### PR TITLE
DEVDOCS-4491: [net new] create checkout token endpoint, Checkouts v3

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-4491]

## What changed?
Added Checkout Token endpoint

## Anything else?

[DEVDOCS-4491]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4491?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/checkouts.v3.yml [hunk 1]
```diff
@@ -7933,6 +7934,60 @@ paths:
...
+          application/json:
+            schema:
+              type: object
+              properties:
+                maxUses:
+                  type: number
+                  example: 1
+                ttl:
+                  type: number
+                  example: 9999999999999
```
Comment (bc-mackxu) [inline reference/checkouts.v3.yml:7984]:
```suggestion
                  example: 86400
```

Could we add a description for the field `ttl`? The number means how many seconds for the token alive and maximum is 2,592,000 (30 days).

**Review by bc-mackxu — DISMISSED** [review]
👏 Nice, thank you Traci

**Comment by markcmurphy** [comment]
> Left-side navigation for this new endpoint needs to be created. How do I create it?

It'll appear automatically. We only need to update the table of contents for new spec files. We will need to create "permanent" human-readable urls and set up redirects, though.

File: reference/checkouts.v3.yml [hunk 2]
```diff
@@ -7933,6 +7934,65 @@ paths:
                 meta: {}
       security:
         - X-Auth-Token: []
+  '/stores/{store_hash}/v3/checkouts/{checkout-id}/token':
```
Comment (slsriehl) [inline reference/checkouts.v3.yml:7937]: standardize `checkout-id` to `checkoutId` throughout.

Comment (bc-traciporter) [reply reference/checkouts.v3.yml:7937]: good catch @slsriehl. I had to create a new branch to merge the changes. (DEVDOCS-4491-feedback)

**Comment by slsriehl** [comment]
superceded by #1012. reopen if needed.

**Pull request closed without merging**